File: C++/Check_Path.cpp
#include <iostream>
#include <vector>
#include <stack>

using namespace std;

void addEdge(vector<vector<int>>& adj, int u, int v) {
    adj[u].push_back(v);
    adj[v].push_back(u); // For an undirected graph
}

bool hasPath(vector<vector<int>>& adj, int src, int dest) {
    vector<bool> visited(adj.size(), false);
    stack<int> stack;

    visited[src] = true;
    stack.push(src);

    while (!stack.empty()) {
        int current = stack.top();
        stack.pop();

        for (int neighbor : adj[current]) {
            if (neighbor == dest) {
                return true; // Path found
            }
            if (!visited[neighbor]) {
                visited[neighbor] = true;
                stack.push(neighbor);
            }
        }
    }

    return false; // No path found
}

File: Java/Check_Path.java
import java.util.ArrayList;
import java.util.List;

public class Graph {
    private int V;
    private List<List<Integer>> adjList;

    public Graph(int V) {
        this.V = V;
        adjList = new ArrayList<>(V);
        for (int i = 0; i < V; i++) {
            adjList.add(new ArrayList<>());
        }
    }

    public void addEdge(int u, int v) {
        adjList.get(u).add(v);
        adjList.get(v).add(u); // For an undirected graph
    }

    public boolean hasPath(int src, int dest) {
        boolean[] visited = new boolean[V];
        List<Integer> queue = new ArrayList<>();

        visited[src] = true;
        queue.add(src);

        while (!queue.isEmpty()) {
            int current = queue.remove(0);

            for (int neighbor : adjList.get(current)) {
                if (neighbor == dest) {
                    return true; // Path found
                }
                if (!visited[neighbor]) {
                    visited[neighbor] = true;
                    queue.add(neighbor);
                }
            }
        }

        return false; // No path found
    }
}
